### ISSUE TITLE
[discussion]Suggestion on categorical probability error message.

### ISSUE THREAD

**Issue by yaowang1111** [pyro-ppl/pyro#3108]
When you provide a categorical distribution probs=torch.tensor([0,0]), it will give exception:
 invalid multinomial distribution (encountering probability entry < 0)

I think this exception might be modified to increase accuracy.  First, change multinomial to categorical may be more accurate; second, <0 is confusing here because we are providing 0, it might be better restated as
invalid categorical distribution(check probability entry values, all entries should be >=0 and their sum >0)
